src/hypothesis/core.py

```diff
@@ -102,8 +114,13 @@ def reify_and_execute(
 ):
     def run(data):
         with BuildContext(data, is_final=is_final):
-            import random as rnd_module
-            rnd_module.seed(0)
+            orig = sys.gettrace()
+            try:  # pragma: no cover
+                sys.settrace(None)
+                import random as rnd_module
+                rnd_module.seed(0)
+            finally:  # pragma: no cover
+                sys.settrace(orig)
```

Comment:
Since you do this kind of thing in three places, it might be worth adding a context manager for it so you can do something like:

```python
with no_tracing():
    import random as rnd_module
    rnd_module.seed(0)
```

Reply:
I thought about it, and I'll probably give it a try, but I'm a little worried about interactions between the extra function calls and `settrace` doing interesting things, so I decided to leave it until I've got the thing actually working.

Reply:
I tried this, and [I did indeed see strange behaviour](https://travis-ci.org/HypothesisWorks/hypothesis-python/builds/275881516). Might just have been my doing something wrong, but I didn't feel like debugging it right at this minute